### PR TITLE
AlertingNG: List saved Alert definitions in Alert Rule list

### LINKED ISSUE #30600: 7.4: Placeholder - re-use the Alerting List view we have today to show list of alert definitions

### LINKED ISSUE #30815: AlertingNG: List alert instances
List alert instances in the instances tab.

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Lists Alert definitions with the Alert rules at `/alerting/list`. 

**Which issue(s) this PR fixes**:

Fixes #30815

**Special notes for your reviewer**:

### CODE REVIEW

**Review by kylebrandt — APPROVED** [review]
I see alert definitions listed:

image

And don't see them when I turn the flag off (as expected). 

Looks like some tests to fix. Don't forget to backport to 7.4.x

File: public/app/features/alerting/components/AlertDefinitionItem.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  search: string;
+}
+
+export const AlertDefinitionItem: FC<Props> = ({ alertDefinition, search }) => {
+  return (
+    <li
+      className={css`
+        width: 100%;
+      `}
+    >
```
Comment (torkelo) [inline public/app/features/alerting/components/AlertDefinitionItem.tsx:20]: this li element seems very unnecessary, we can just change the alert-rule-list to display flex or use VerticalGroup

**Comment by torkelo** [comment]
Nice, like that that they share the same list, will make migrations & having two alerting engines easier for the user

File: public/app/features/alerting/components/AlertDefinitionItem.tsx (hunk 1 above)
Comment (peterholmberg) [reply public/app/features/alerting/components/AlertDefinitionItem.tsx:20]: yeah, the ol/li setup looked weird to me.

**Pull request merged**